src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php

```diff
@@ -0,0 +1,242 @@
+        RepositoryInterface $shippingCategoryRepository,
+        RepositoryInterface $channelRepository,
+        RepositoryInterface $localeRepository
+    ) {
+        $this->productFactory = $productFactory;
+        $this->variantGenerator = $variantGenerator;
+        $this->localeRepository = $localeRepository;
+
+        $this->faker = \Faker\Factory::create();
+        $this->optionsResolver =
+            (new OptionsResolver())
+                ->setDefault('name', function (Options $options) {
+                    return $this->faker->words(3, true);
+                })
+                ->setDefault('code', function (Options $options) {
+                    return StringInflector::nameToCode($options['name']);
+                })
+                ->setDefault('enabled', function (Options $options) {
+                    return $this->faker->boolean(90);
+                })
+                ->setAllowedTypes('enabled', 'bool')
+                ->setDefault('short_description', function (Options $options) {
+                    return $this->faker->paragraph;
+                })
+                ->setDefault('description', function (Options $options) {
+                    return $this->faker->paragraphs(3, true);
+                })
+                ->setAllowedTypes('enabled', 'bool')
+                ->setDefault('main_taxon', LazyOption::randomOne($taxonRepository))
+                ->setAllowedTypes('main_taxon', ['null', 'string', TaxonInterface::class])
+                ->setNormalizer('main_taxon', LazyOption::findOneBy($taxonRepository, 'code'))
+                ->setDefault('product_archetype', LazyOption::randomOne($productArchetypeRepository))
+                ->setAllowedTypes('product_archetype', ['null', 'string', ArchetypeInterface::class])
+                ->setNormalizer('product_archetype', LazyOption::findOneBy($productArchetypeRepository, 'code'))
+                ->setDefault('shipping_category', LazyOption::randomOne($shippingCategoryRepository))
+                ->setAllowedTypes('shipping_category', ['null', 'string', ShippingCategoryInterface::class])
+                ->setNormalizer('shipping_category', LazyOption::findOneBy($shippingCategoryRepository, 'code'))
+                ->setDefault('taxons', LazyOption::randomOnes($taxonRepository, 3))
+                ->setAllowedTypes('taxons', 'array')
+                ->setNormalizer('taxons', LazyOption::findBy($taxonRepository, 'code'))
+                ->setDefault('channels', LazyOption::randomOnes($channelRepository, 3))
+                ->setAllowedTypes('channels', 'array')
+                ->setNormalizer('channels', LazyOption::findBy($channelRepository, 'code'))
+                ->setDefault('product_attributes', [])
+                ->setAllowedTypes('product_attributes', 'array')
+                ->setNormalizer('product_attributes', function (Options $options, array $productAttributesValues) use ($productAttibuteValueFactory) {
```

Comment:
i'd say we need some kind of ordering/sorting for all these calls so it'd be easier to read. WDYT?